pkg/engine/daemon.go

```diff
@@ -93,26 +99,31 @@ func getPath(instructions string) (string, string) {
 	return strings.TrimSpace(rest), strings.TrimSpace(value)
 }
 
-func (e *Engine) startDaemon(_ context.Context, tool types.Tool) (string, error) {
-	e.Ports.daemonLock.Lock()
-	defer e.Ports.daemonLock.Unlock()
+func (e *Engine) startDaemon(tool types.Tool) (string, error) {
+	ports.daemonLock.Lock()
+	defer ports.daemonLock.Unlock()
 
 	instructions := strings.TrimPrefix(tool.Instructions, types.DaemonPrefix)
 	instructions, path := getPath(instructions)
 	tool.Instructions = types.CommandPrefix + instructions
 
-	port, ok := e.Ports.daemonPorts[tool.ID]
+	port, ok := ports.daemonPorts[tool.ID]
 	url := fmt.Sprintf("http://127.0.0.1:%d%s", port, path)
```

Comment:
Not sure if there would be any use for this (and it's not part of your PR anyway), but can this be made configurable to listen on other interfaces as well?

Reply:
We can definitely discuss that!